src/foam/lib/formatter/JSONFObjectFormatter.java

```diff
@@ -339,8 +339,7 @@ public void outputDelta(FObject oldFObject, FObject newFObject, ClassInfo defaul
     }
     addInnerNewline();
     PropertyInfo id = (PropertyInfo) newInfo.getAxiomByName("id");
-    outputProperty(newFObject, id);
-
+    if ( ! id.getStorageTransient() ) outputProperty(newFObject, id);
```

Comment:
If we don't output the id, then are we going to get an extra "," output on line 346?

Reply:
Fixed in https://github.com/foam-framework/foam2/pull/4229/commits/fb36c41ba46989d1b99a2fdcdf9324a9684cf5c0.